src/sync/installation.ts

```diff
@@ -110,7 +110,19 @@ export const updateJobStatus = async (
 	const status = isComplete ? "complete" : "pending";
 
 	logger.info({ status }, "Updating job status");
-	await updateRepo(subscription, repositoryId, { [getStatusKey(task)]: status });
+
+	const values: { [x: string]: string | Date} = { [getStatusKey(task)]: status };
+
+	if (isComplete && task === "commit" && data.commitsFromDate) {
+		const repoSync = await RepoSyncState.findByRepoId(subscription, repositoryId);
+		const commitsFromDate =  new Date(data.commitsFromDate);
+		// Set commitsFromDate in RepoSyncState only if its the later date
+		if (repoSync &&	(!repoSync.commitFrom || repoSync.commitFrom.getTime() > commitsFromDate.getTime())) {
```

Comment:
maybe this is easier to read?
```
const newCommitsFromDate =  new Date(data.commitsFromDate);
(repoSync && repoSync.commitFrom && repoSync.commitFrom.getTime() > newCommitsFromDate.getTime()
```

Reply:
One case missing in your expression when repoSync.commitFrom is null